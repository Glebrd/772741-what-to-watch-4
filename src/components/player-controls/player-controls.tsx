import * as React from "react";
import withProgress from "../../hocs/with-progress/with-progress";

const PlayerControls = (props) => {
  const {progressValue, remainingTime} = props;
  return (
    <div className="player__controls-row">
      <div className="player__time">
        <progress className="player__progress" max="100"></progress>
        <div style={{left: `${progressValue}%`}} className="player__toggler">Toggler</div>
      </div>
      <div className="player__time-value">{remainingTime}</div>
    </div>
  );
};

// PlayerControls.propTypes = {
//   progressValue: PropTypes.number,
//   remainingTime: PropTypes.string,
// };

export {PlayerControls};
export default withProgress(PlayerControls);
