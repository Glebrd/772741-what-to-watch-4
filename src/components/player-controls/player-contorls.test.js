import React from "react";
import renderer from "react-test-renderer";
import {PlayerControls} from "./player-controls";

const progressValue = 30;
const remainingTime = 120;

it(`Player controls renders`, () => {
  const tree = renderer
    .create(
        <PlayerControls
          progressValue={progressValue}
          remainingTime={remainingTime}
        />
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
