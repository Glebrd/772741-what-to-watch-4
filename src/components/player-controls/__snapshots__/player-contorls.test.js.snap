// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player controls renders 1`] = `
<div
  className="player__controls-row"
>
  <div
    className="player__time"
  >
    <progress
      className="player__progress"
      max="100"
    />
    <div
      className="player__toggler"
      style={
        Object {
          "left": "30%",
        }
      }
    >
      Toggler
    </div>
  </div>
  <div
    className="player__time-value"
  >
    120
  </div>
</div>
`;
